fido/Crypto/Fido2/Attestation/AndroidKey/Statement.hs

```diff
@@ -0,0 +1,86 @@
+-- | Implements step 1-3 of the verification procedure of chapter 8.4
+module Crypto.Fido2.Attestation.AndroidKey.Statement
+  ( Stmt
+      ( Stmt,
+        alg,
+        sig,
+        x5c,
+        attExt
+      ),
+    ExtAttestation
+      ( ExtAttestation,
+        attestationChallenge,
+        softwareEnforced,
+        teeEnforced
+      ),
+    decode,
+  )
+where
+
+import Codec.CBOR.Decoding (Decoder)
+import Codec.CBOR.Term (Term (TBytes, TInt, TList, TString))
+import Crypto.Fido2.PublicKey (COSEAlgorithmIdentifier, toAlg)
+import Crypto.Hash (Digest, SHA256, digestFromByteString)
+import Data.ASN1.Parse (ParseASN1, getNext, getNextContainer, onNextContainer, runParseASN1)
+import Data.ASN1.Types (ASN1 (OctetString), ASN1ConstructionType (Sequence))
+import Data.Bifunctor (Bifunctor (first))
+import Data.ByteString (ByteString)
+import qualified Data.Map as Map
+import Data.X509 (Extension, extensionGetE)
+import qualified Data.X509 as X509
+
+-- androidStmtFormat (https://www.w3.org/TR/webauthn-2/#sctn-android-key-attestation)
+data Stmt = Stmt
+  { alg :: COSEAlgorithmIdentifier,
+    sig :: ByteString,
+    x5c :: X509.SignedExact X509.Certificate,
+    -- | Holds the parsed attestation extension of the above X509 certificate
+    -- Not part of the spec, but prevents parsing in the AndroidKey.verify function
+    attExt :: ExtAttestation
+  }
+  deriving (Show)
+
+data ExtAttestation = ExtAttestation
+  { attestationChallenge :: Digest SHA256,
+    softwareEnforced :: [ASN1],
+    teeEnforced :: [ASN1]
+  }
+  deriving (Show)
+
+instance Extension ExtAttestation where
+  extOID = const [1, 3, 6, 1, 4, 1, 11129, 2, 1, 17]
+  extHasNestedASN1 = const True
+  extEncode = error "Can no encode the parsed ExtAttestation to a valid [ASN1] because most fields are dropped during parsing."
+  extDecode asn1 = first (const $ fail "could not decode ASN1 attestation extension") $ runParseASN1 decodeExtAttestation asn1
+    where
+      decodeExtAttestation :: ParseASN1 ExtAttestation
+      decodeExtAttestation = onNextContainer Sequence decodeKeyDescription
+
+      decodeKeyDescription :: ParseASN1 ExtAttestation
+      decodeKeyDescription = do
+        _attestationVersion <- getNext
```

Comment:
We should have a comment here explaining why it's fine to not look at the version. The reason is that all 3 versions of the existing schema versions don't differ in anything we're interested in. If we were to parse all fields we would have to look at it